### ISSUE TITLE
Force-complete not really completing (sometimes?)

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#6710]
As reported in this elog:
https://cms-logbook.cern.ch/elog/Workflow+processing/23460

the force-completion didn't really work, these are my observations of the system:
- WQE were properly canceled by GQ, see some lines from the GQ logs in [1]
- then the request was "finished" and "deleted" in GQ, see logs in [1]
- meanwhile, vocms0309 that had previous WQE acquired, was still creating subscriptions
  and injecting data into WMBS, see logs in [2]
- looking at local couch in vocms0309 (workqueue db), I see 362 Running WQE out of 363, which is clearly wrong.

The main issue here seems to be coming from the fact that these WQE didn't replicate from GQ to LQ, so their status is inconsistent and the agent does not kill anything. I was considering to make a component responsible for these actions on the agent side, but then we will most likely have doc conflicts in couch, right?

Jean-Roch also tells me they have been using it extensively, proving there are no problems, in general.

Anyways, just opening a discussion here :)

[1]

```
2016-03-15 12:25:34,243 INFO     WorkQueue: Canceling work for workflow(s): set(['pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225'])
2016-03-15 12:26:20,388 INFO     WorkQueue: Cancel-ed element(s) ['01793b7e9928ab8b9c2f1104352315a9', '018a2a1b184cf8b086c1d0f582dcd1da', '01baad5b00e6a65faf1da73c854fe906',

2016-03-15 12:27:07,132 INFO     WorkQueue: CancelRequest-ed element(s) ['0024c999cc572ec2003f45086a600474', '0025e383d56665ee1be5dc7179617f81', '003a2ee27350273c900296f31114c6a8',

2016-03-15 17:47:00,670 INFO     WorkQueue: Canceling work for workflow(s): set(['pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225'])
2016-03-15 17:47:36,643 INFO     WorkQueue: Canceling work for workflow(s): set(['pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225'])

2016-03-15 17:55:59,044 INFO     WorkQueue: Request pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225 finished ({'Status': 'Canceled', 'PercentSuccess': 32, 'PercentComplete': 60})

2016-03-15 18:07:02,779 INFO     WorkQueue: Canceling work for workflow(s): set(['pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225'])
2016-03-15 18:07:12,018 INFO     WorkQueue: Deleting request "pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225" as it is Canceled
```

[2]

```
2016-03-15 14:15:04,219:INFO:WMBSHelper:"pdmvserv_FSQ-RunIISpring15PrePremix-00004_00008_v2_AVE_50_BX_25ns_160309_183357_8225" Injecting production 1:646001:52972001 - 1:646500:53013000 (run:lumi:event) i
nto wmbs
```

**Comment by amaltaro** [comment]
BTW, I restarted both couch and WQ in vocms0309 (to get the replication moving again), and the replication progress is 100% and there are 9 missing revisions, see
```
{"pid":"<0.213.0>","checkpoint_interval":5000,"checkpointed_source_seq":0,"continuous":true,"doc_id":"98bdf41e642a79faec2cfae30dfd256b","doc_write_failures":0,"docs_read":9,"docs_written":9,"missing_revisions_found":9,"progress":100,"replication_id":"32faa271c366e3142fb19b9a06f390d4+continuous","revisions_checked":2427,"source":"https://cmsweb.cern.ch/couchdb/workqueue/","source_seq":11503596,"started_on":1458130594,"target":"http://localhostBLAH/workqueue_inbox/","type":"replication","updated_on":1458134506},
```

and still, WQE in status Running.

**Comment by amaltaro** [comment]
Should have been fixed by #7442
